src/components/camera/CameraStream.jsx

```diff
@@ -13,44 +13,73 @@ import "./CameraStream.css";
-  let newWindow = window.open("", "", "width=500,height=500");
-  newWindow.document.body.style.margin = "0";
-  newWindow.document.title = cameraTitle + " Stream";
-  newWindow.document.body.innerHTML = `
-    <div style="background-color:black;width:100%;height:100%;font-family:Arial,Helvetica,sans-serif;display:flex;justify-content:center">
-      <div style="z-index:100;position:absolute;width:100%;font-size:30px;font-weight:bold;text-align:center;color:white;padding-top:5px;padding-bottom:5px;background-color:#00000066;">${cameraTitle}</div>
-      <div id="ext-fps" style="z-index:101;position:absolute;top:5;left:5;font-size:15px;font-weight:bold;color:red;">FPS: N/A</div>
-      <canvas id="ext-vid" style="z-index:1;border:none;display:block;margin:auto 0;"></canvas>
-      </div>
-    </div>`;
-  let canvas = newWindow.document.querySelector('#ext-vid');
-  let context = canvas.getContext('2d');
-  let aspectRatio = document.querySelector(`#${cameraName}-player`).videoHeight / document.querySelector(`#${cameraName}-player`).videoWidth;
+async function createPopOutWindow(cameraTitle, cameraName, unloadCallback, video_width, video_height) {
+  let newWindow = window.open("/camera/cam_popout.html", "", "width=500,height=500");
+
+  const returnPromise = new Promise((resolve, reject) => {
+    newWindow.onload = () => {
+      newWindow.document.title = `${cameraTitle} Stream`;
+      newWindow.document.querySelector('#ext-title').innerText = cameraTitle;
+      newWindow.document.querySelector('#ext-download-button').setAttribute("onclick", `download("${cameraTitle}", ${video_width}, ${video_height})`);
+
+      let canvas = newWindow.document.querySelector('#ext-vid');
+      let context = canvas.getContext('2d');
+      let aspectRatio = document.querySelector(`#${cameraName}-player`).videoHeight / document.querySelector(`#${cameraName}-player`).videoWidth;
+      canvas.width = aspectRatio * 400;
+      canvas.height = 400;
+      context.fillStyle = "blue";
+      context.fillRect(0, 0, canvas.width, canvas.height);
+
+      window.onunload = () => {
+        if (newWindow && !newWindow.closed) {
+            newWindow.close();
+        }
+      };
 
-  canvas.width = aspectRatio * 400;
-  canvas.height = 400;
-  context.fillStyle = "blue";
-  context.fillRect(0, 0, canvas.width, canvas.height);
+      newWindow.onbeforeunload = unloadCallback;
 
-  window.onunload = () => {
-    if (newWindow && !newWindow.closed) {
-        newWindow.close();
+      let output = {
+        popout: newWindow,
+        canvas: canvas,
+        context: context,
+        aspectRatio: aspectRatio
+      };
+      resolve(output); 
     }
-  };
+  });
+
+  return returnPromise;
+}
+
+/**
+ * Takes
+ *    video: HTMLVideoElement representing the video tag which should be processed.
+ *    cameraTitle: name of the camera, used for the filename.
+ */
+function downloadCurrentFrame(video, cameraTitle) {
```

Comment:
Looks like this function has a lot of duplication with `download()` in `cam_popout.js`. I assume it's probably hard to remove that duplication, since one is in the react codebase and the other is outside, but can you at least leave a comment in each function pointing to the other one? That way when we modify it in the future we know to modify both.